HopsanGUI/HcomHandler.cpp

```diff
@@ -7134,6 +7158,211 @@ void HcomHandler::evaluateExpression(QString expr, VariableType desiredType)
             return;
         }
     }
+    else if(desiredType != Scalar && desiredType != DataVector && isHcomFunctionCall("expr", expr)) {
+       QStringList args = extractFunctionCallExpressionArguments(expr);
+       if(args.size() != 1) {
+           HCOMERR("Wrong number of arguments for expr function.\n"+mLocalFunctionDescriptions.find("ddt").value().second);
+           mAnsType = Undefined;
+           return;
+       }
+       bool ok;
+       mAnsExpression = SymHop::Expression(args[0], &ok, SymHop::Expression::NoSimplifications);
+       if(!ok) {
+           for(const auto &msg : mAnsExpression.readErrorMessages()) {
+               HCOMERR(msg);
+           }
+           mAnsType = Undefined;
+           return;
+       }
+       mAnsType = Expression;
+       return;
+    }
+    else if(isHcomFunctionCall("eval", expr)) {
+        QStringList args = extractFunctionCallExpressionArguments(expr);
+        if(args.size() != 1) {
+            HCOMERR("Wrong number of arguments for eval function.\n"+mLocalFunctionDescriptions.find("ddt").value().second);
+            mAnsType = Undefined;
+            return;
+        }
+        if(mLocalExpressions.contains(args[0])) {
+            bool ok;
+            double value = mLocalExpressions[args[0]].evaluate(mLocalVars,&mLocalFunctionoidPtrs, &ok);
+            if(ok) {
+                mAnsScalar = value;
+                mAnsType = Scalar;
+                return;
+            }
+            else {
+                HCOMERR("Failed to evaluate expression \""+args[0]+"\".");
+                mAnsType = Undefined;
+                return;
+            }
+        }
+        else {
+            HCOMERR("Local expression \""+args[0]+"\" not found.");
+            mAnsType = Undefined;
+            return;
+        }
+    }
+    else if(isHcomFunctionCall("der", expr) || isHcomFunctionCall("factor", expr)) {
+        QStringList args = extractFunctionCallExpressionArguments(expr);
+        if(args.size() != 2) {
+            HCOMERR("Wrong number of arguments for \""+getFunctionName(expr)+"\" function (should be 2)");
+            mAnsType = Undefined;
+            return;
+        }
+        SymHop::Expression expr1, expr2;
+        evaluateExpression(args[0]);
+        if(mAnsType != Expression) {
+            HCOMERR("Failed to evaluate expression \""+args[0]+"\".");
+            mAnsType = Undefined;
+            return;
+        }
+        expr1 = mAnsExpression;
+        evaluateExpression(args[1]);
+        if(mAnsType != Expression) {
+            HCOMERR("Failed to evaluate expression \""+args[1]+"\".");
+            mAnsType = Undefined;
+            return;
+        }
+        expr2 = mAnsExpression;
+        if(getFunctionName(expr) == "der") {
+            bool ok;
+            mAnsExpression = mLocalExpressions[args[0]].derivative(mAnsExpression,ok);
```

Comment:
Should we not use the expr1 and expr2  in some way this section of the code ?

Reply:
Yes, that's true. It is fixed now.